core/transaction_pool/impl/transaction_pool_impl.hpp

```diff
@@ -90,7 +88,8 @@ namespace kagome::transaction_pool {
     std::list<WaitingTransaction> waiting_queue_;
 
     // tags provided by imported transactions
-    std::map<primitives::TransactionTag, common::Hash256> provided_tags_;
+    std::map<primitives::TransactionTag, std::optional<common::Hash256>>
```

Comment:
why map of optionals?

Reply:
The origin of some tags is unknown, for example ones provided during pruneTag operation.